src/formats/PDB.cpp

```diff
@@ -639,6 +639,102 @@ static std::string to_pdb_index(uint64_t i) {
     }
 }
 
+struct ResidueInformation {
+    std::string atom_hetatm{ "HETATM" };
+    std::string resname { "XXX" };
+    std::string resid { "  -1" };
+    std::string chainid { "X" };
+    std::string inscode { " " };
+    std::string comp_type { "" };
+};
+
+static ResidueInformation get_residue_strings(optional<const Residue&> residue_opt, uint64_t& max_resid) {
+
+    ResidueInformation res_info;
+
+    if (!residue_opt) {
+        auto value = max_resid++;
+        if (value < 9999) {
+            res_info.resid = to_pdb_index(value);
+        }
+        else {
+            res_info.resid = "  -1";
+        }
+
+        return res_info;
+    }
+
+    auto residue = *residue_opt;
+
+    res_info.resname = residue.name();
+    if (residue.get<Property::BOOL>("is_standard_pdb").value_or(false)) {
+        // only use ATOM if the residue is standardized
+        res_info.atom_hetatm = "ATOM  ";
+    }
+
+    if (res_info.resname.length() > 3) {
+        warning("PDB writer", "residue '{}' name is too long, it will be truncated", res_info.resname);
+        res_info.resname = res_info.resname.substr(0, 3);
+    }
+
+    if (residue.id()) {
+        auto value = residue.id().value();
+        if (value > 9999) {
+            auto negative_conversion = std::to_string(static_cast<int64_t>(value));
+            if (negative_conversion.length() <= 4) {
+                warning("PDB writer", "it appears {} is actually negative, treating as {}", value, negative_conversion);
```

Comment:
Are residue id potentially negative in standard PDB? We could change how we store the id if this is the case

Reply:
Yes it will =). But that's fine, the next release will be a breaking one. I have other improvement idea for residues, I'll write an issue to discuss them!

Reply:
Should negative resid also be handled by mmCIF & MMTF parsers?

Reply:
Sounds good! I think MMTF supports them already since it uses signed integers to store residue IDs, but mmCIF definetly should.